### ISSUE TITLE
bug: FP8 cache warmup corrupts workspace — parity gate fails (cosine -0.28)

### ISSUE THREAD

**Issue by noahgift** [paiml/realizar#181]
## Summary

After FP8 cache warmup (1472 MB), the parity gate's forward pass produces
garbage logits (cosine -0.28, threshold 0.98). With `FP8_PREFILL=0`, the
gate PASSES and serve works correctly.

## Evidence (Yoga RTX 4060, realizr 0.8.3)

| Condition | Parity Gate | Cosine |
|-----------|-------------|--------|
| `FP8_PREFILL=0` | **PASS** | serves at 149.8 tok/s |
| `FP8_PREFILL=1` (default sm_89) | **FAIL** | -0.28 |
| `SKIP_PARITY_GATE=1` + FP8 | Skipped | serves at 132.5 tok/s (correct output) |

Note: with `SKIP_PARITY_GATE=1`, FP8-enabled serve produces correct tokens.
So the FP8 workspace corruption is transient — it only affects the FIRST
forward pass after warmup.

## Five-Whys

1. Why gate fails? → Cosine -0.28 (garbage GPU logits)
2. Why garbage? → Forward pass runs after FP8 cache warmup
3. Why corrupt? → FP8 warmup allocates 1472 MB + cuBLASLt JIT
4. Why invalidate? → Workspace buffers stale after FP8 alloc
5. **Root cause: workspace not reinitialized after FP8 cache warmup**

## Contract

`fp8-warmup-workspace-v1`: After FP8 cache warmup completes,
`force_workspace_reinit()` MUST be called before any forward pass.
The parity gate's `forward_gpu_resident()` relies on workspace
buffers that FP8 allocation may have relocated.

## Fix

In `preload_and_verify()` (mod.rs:227), after FP8 warmup:
```rust
self.executor.force_workspace_reinit();
```

## Falsification

If `force_workspace_reinit()` after FP8 warmup makes the parity gate
pass, root cause confirmed. If it still fails, the corruption is deeper.
